### PR TITLE
ci: wrap yarn into a retrying loop

### PR DESCRIPTION
**Pull request by jhchabran**
_Behold, duct tape to the rescue._

This wraps `yarn install` commands within a loop that check for the presence of "unexpected errors ..." that are announcing that some network / 500 error / whatever happened when requesting stuff on the registry. 

If such an error is found, the script will run yarn again. If we did not find such an error, the script will simply exit, like it would do without the loop. 

If we go over `$MAX_TIMEOUT_IN_SECONDS` the script will simply abort. 

@sourcegraph/frontend-platform  I know that a possible fix is coming in a few weeks, but this could at least stop the bleeding until then. 

## Test plan

main-dry-run

To test this locally, I botched the registry url in `yarn.lock` to reproduce the kind of errors we've been seeing in CI. Remember to wipe your `./node_modules` folder before doing so, otherwise it won't be taken in account.

### CODE REVIEW

File: dev/ci/yarn-test-separate.sh [hunk 1]
```diff
@@ -3,13 +3,15 @@
...
+./dev/ci/yarn-install-with-retry.sh
+
+# Save the absolute path to the script before changing directories.
+abs_yarn="$(pwd)/dev/ci/yarn-install-with-retry.sh"
 
 cd "$1"
 echo "--- yarn"
 # mutex is necessary since CI runs various yarn installs in parallel
-yarn --mutex network --frozen-lockfile --network-timeout 60000 --silent
+"$abs_yarn"
```
Comment (jhchabran) [inline dev/ci/yarn-test-separate.sh:14]: Yeah, not the prettiest, it'd be better for this script to be in the `$PATH`. That'll do for now.

**Review by burmudar — APPROVED** [review]

**Review by bobheadxi — APPROVED** [review]

File: dev/ci/yarn-test-separate.sh (hunk 1 above)
Comment (marekweb) [reply dev/ci/yarn-test-separate.sh:14]: I suppose we can remove the `# mutex is necessary` comment here as well

File: dev/licenses.sh [hunk 2]
```diff
@@ -29,6 +29,7 @@ trap cleanup EXIT
 go mod tidy
 go mod vendor # go mod download does not work with license_finder
 yarn --mutex network --frozen-lockfile
+./dev/ci/yarn-install-with-retry.sh
```
Comment (marekweb) [inline dev/licenses.sh:32]: Should the original `yarn` call be removed here?

Comment (jhchabran) [reply dev/licenses.sh:32]: Yes, missed it, thank you!

**Review by kalanchan — APPROVED** [review]

**Pull request merged**